### PR TITLE
[image-picker][ios] Add possibility to choose presentation style

### LINKED ISSUE #13614: ImagePicker.launchCameraAsync does not open in the full screen on iOS devises
### Summary

After upgrading to SDK 41, the Camera view does not open to the full screen with **ImagePicker.launchCameraAsync** on iOS devices any more .
On iPad it opens in the center of the screen overlaying the screen it was opened from, which looks very weird.
on iPhone, it also looks like an overlay on top of the parent screen, and you can see the portion of the parent header.
Another indicator that it does not open to the full screen, it that there is a grey shaded area blow the 1x, 2x zoom indicator. If you drag the top portion (the one that has flash cam control) up, the grey area under the zoom indicator disappears (slides down). When you release the finger, everything snaps back to previous state as shown on the attachment photo. The gray area is blocking part of the photo view, which makes it harder to understand the boundaries of the photo being taken. It also gets in a way of the zoom control working properly – definitely something is not working the way it was intended.
57fa9f521f896ca8353581ed7458a5ce9414fb5e

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

41, 42

### Environment

  Expo CLI 4.7.2 environment info:
    System:
      OS: macOS 11.4
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 15.5.0 - ~/.nvm/versions/node/v15.5.0/bin/node
      Yarn: 1.22.10 - /usr/local/bin/yarn
      npm: 7.19.1 - ~/.nvm/versions/node/v15.5.0/bin/npm
      Watchman: 2021.06.07.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
      Android SDK:
        API Levels: 23, 26, 28, 29, 30
        Build Tools: 28.0.3, 29.0.2, 30.0.2
        System Images: android-29 | Intel x86 Atom, android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-29 | Google APIs Intel x86 Atom_64, android-29 | Google Play Intel x86 Atom, android-29 | Google Play Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom
        Android NDK: 22.0.7026061
    IDEs:
      Android Studio: 4.1 AI-201.8743.12.41.6953283
      Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~42.0.1 => 42.0.1 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: https://github.com/expo/react-native/archive/sdk-42.0.0.tar.gz => 0.63.2 
      react-native-web: ~0.13.12 => 0.13.18 
    npmGlobalPackages:
      expo-cli: 4.7.2
    Expo Workflow: managed

### Reproducible demo or steps to reproduce from a blank project

Barebones project was created to demonstrate the issue https://github.com/echowaves/CameraBug
Just install it and run `yarn; yarn start`, then open on your iPhone or iPad with expo.
Click a button (the only button on a screen) to launch a camera view and see that it does not open to the full screen.

Here is the line where the `ImagePicker.launchCameraAsync` is invoked.

https://github.com/echowaves/CameraBug/blob/cff7c0d4d583fa58a644ffcc31895005c7c0e82b/App.js#L19

### PR DESCRIPTION
**Pull request by mstach60161**
# Why

- Fixes #13614 

# How

- Add possibility to choose any presentation style.

# Test Plan

Native component list.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/imagepicker.md [hunk 1]
```diff
@@ -152,6 +152,7 @@ Display the system UI for choosing an image or a video from the phone's library.
   - **mediaTypes (_[ImagePicker.MediaTypeOptions](#imagepickermediatypeoptions)_)** -- Choose what type of media to pick. Defaults to `ImagePicker.MediaTypeOptions.Images`.
   - **allowsEditing (_boolean_)** -- Whether to show a UI to edit the image/video after it is picked. Images: On Android the user can crop and rotate the image and on iOS simply crop it. Videos: On iOS user can trim the video. Defaults to `false`.
   - **allowsMultipleSelection (_boolean_)** -- (Web only) Whether or not to allow selecting multiple media files at once.
+  - **presentationStyle (_[ImagePicker.UIImagePickerPresentationStyle](#imagePickeruiimagepickerpresentationstyle)_)** -- (IOS only) Choose [presentation style](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621355-modalpresentationstyle?language=objc) to customize view during picking photo/video. Defaults to `ImagePicker.UIImagePickerPresentationStyle.Automatic`.
```
Comment (bbarthec) [inline docs/pages/versions/unversioned/sdk/imagepicker.md:155]:
```suggestion
  - **presentationStyle (_[ImagePicker.UIImagePickerPresentationStyle](#imagePickeruiimagepickerpresentationstyle)_)** -- (iOS only) Choose [presentation style](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621355-modalpresentationstyle?language=objc) to customize view during picking photo/video. Defaults to `ImagePicker.UIImagePickerPresentationStyle.Automatic`.
```

File: docs/pages/versions/unversioned/sdk/imagepicker.md [hunk 2]
```diff
@@ -194,6 +195,7 @@ Display the system UI for taking a photo with the camera. Requires `Permissions.
 
   - **mediaTypes (_[ImagePicker.MediaTypeOptions](#imagepickermediatypeoptions)_)** -- Choose what type of media to pick. Defaults to `ImagePicker.MediaTypeOptions.Images`.
   - **allowsEditing (_boolean_)** -- Whether to show a UI to edit the image after it is picked. On Android the user can crop and rotate the image and on iOS simply crop it. Defaults to `false`.
+  - **presentationStyle (_[ImagePicker.UIImagePickerPresentationStyle](#imagePickeruiimagepickerpresentationstyle)_)** -- (IOS only) Choose [presentation style](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621355-modalpresentationstyle?language=objc) to customize view during taking photo/video. Defaults to `ImagePicker.UIImagePickerPresentationStyle.Automatic`.
```
Comment (bbarthec) [inline docs/pages/versions/unversioned/sdk/imagepicker.md:198]:
```suggestion
  - **presentationStyle (_[ImagePicker.UIImagePickerPresentationStyle](#imagePickeruiimagepickerpresentationstyle)_)** -- (iOS only) Choose [presentation style](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621355-modalpresentationstyle?language=objc) to customize view during taking photo/video. Defaults to `ImagePicker.UIImagePickerPresentationStyle.Automatic`.
```

File: packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m [hunk 3]
```diff
@@ -208,8 +208,12 @@ - (void)launchImagePicker:(EXImagePickerTarget)target
     }
     
     self.picker.videoMaximumDuration = videoMaxDuration;
-    
-    self.picker.modalPresentationStyle = UIModalPresentationPageSheet;
+
+    if (self.options[@"presentationStyle"]) {
+      self.picker.modalPresentationStyle = (UIModalPresentationStyle) [self.options[@"presentationStyle"] intValue];
+    } else {
+        self.picker.modalPresentationStyle = UIModalPresentationAutomatic;
```
Comment (bbarthec) [inline packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m:215]:
```suggestion
      self.picker.modalPresentationStyle = UIModalPresentationAutomatic;
```

**Review by bbarthec — APPROVED** [review]
Looks nice 👍 
Apply suggestions and we're ready to merge it 👍

**Review by lukmccall — APPROVED** [review]
LGTM 💪

**Pull request merged**